package.json

```diff
@@ -16,7 +16,7 @@
   "scripts": {
     "build": "npm run --workspaces --if-present build && npm run build:types",
     "build:types": "tsc -b",
-    "clean:types": "tsc -b --clean",
+    "clean:types": "git clean -fx -e \"**/node_modules/**\" \"packages/*/types/*.d.ts*\" \"*.tsbuildinfo\" || tsc -b --clean",
```

Comment:
nice that git emits the cleaned files, was this the reason

Reply:
I changed it because `tsc -b --clean` only cleans files generated by `tsc -b`. if you are switching branches often, you may have files hanging around that were generated from a different state that the current configuration doesn't know about. so `git clean` is really the only way to be sure.  however, `git clean` will exit with a nonzero code _if there's nothing to clean_; hence `|| tsc -b --clean` (which I hope works on windows).